core/src/main/java/org/infinispan/topology/LocalTopologyManagerImpl.java

```diff
@@ -217,7 +219,22 @@ public void run() {
          }
       }, null);
       // Clear any finished tasks from the completion service's queue to avoid leaks
-      cacheStatus.getTopologyUpdatesCompletionService().drainCompletionQueue();
+      List<? extends Future<Void>> futures = cacheStatus.getTopologyUpdatesCompletionService().drainCompletionQueue();
+      boolean interrupted = false;
+      for (Future<Void> future : futures) {
+         // These will not block since they are already completed
+         try {
+            future.get();
+         } catch (InterruptedException e) {
+            // This shouldn't happen as each task is complete
+            interrupted = true;
+         } catch (ExecutionException e) {
+            log.topologyUpdateError(cacheTopology.getTopologyId(), e.getCause());
```

Comment:
Same thing here, we will receive a new topology and we will call `drainCompletionQueue()` as long as everything goes fine, but there's no guarantee that it will be called when things go wrong.